packages/core/src/api/oauth2.ts

```diff
@@ -121,4 +123,31 @@ export class OAuth2API {
 			signal,
 		}) as Promise<RESTGetAPIOAuth2CurrentAuthorizationResult>;
 	}
+
+	/**
+	 * Revokes an OAuth2 token
+	 *
+	 * @see {@link https://discord.com/developers/docs/topics/oauth2#authorization-code-grant-token-revocation-example}
+	 * @param applicationId - The application id
+	 * @param applicationSecret - The application secret
+	 * @param body - The body of the token revocation request
+	 * @param options - The options for the token revocation request
+	 */
+	public async revokeToken(
+		applicationId: Snowflake,
+		applicationSecret: string,
+		body: RESTPostOAuth2TokenRevocationQuery,
+		{ signal }: Pick<RequestData, 'signal'> = {},
+	) {
+		await this.rest.post(Routes.oauth2TokenRevocation(), {
+			body: makeURLSearchParams(body),
+			passThroughBody: true,
+			headers: {
+				Authorization: `Basic ${btoa(`${applicationId}:${applicationSecret}`)}`,
```

Comment:
The original PR indeed had this added with node:buffer, but i was requested to change to btoa.

Reply:
making a spec feature "legacy" is completely unhinged if the only alternative is a runtime-specific API

Reply:
It isn't the only alternative, `btoa` and `atob` are sort of discouraged because it takes in ASCII strings rather than UCS-2/UTF-16 which JavaScript uses. See the [Unicode Problem](https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem) documented in MDN.

That being said, the input here is entirely ASCII (7-bits, or within `0x00..0x7A`) so this specific usage is fine as long as Discord keeps both strings ASCII. Resolving.